sympy/solvers/tests/test_solveset.py

```diff
@@ -1968,17 +1967,16 @@ def test_solve_lambert():
         FiniteSet(LambertW(3*exp(-LambertW(3))))
 
     assert solveset_real(x*log(x) + 3*x + 1, x) == \
-        Intersection(Interval(0, oo), FiniteSet(exp(-3 + LambertW(-exp(3)))))
+        FiniteSet(exp(-3 + LambertW(-exp(3))))
     # issue 4271
-    assert solveset_real((a/x + exp(x/2)).diff(x, 2), x) == Intersection(
-        Interval(0, oo), FiniteSet(6*LambertW((-a)**(S(1)/3)/3))) - FiniteSet(0)
+    assert solveset_real((a/x + exp(x/2)).diff(x, 2), x) == \
+        FiniteSet(6*LambertW((-a)**(S(1)/3)/3)) - FiniteSet(0)
```

Comment:
There are two solutions when `a=1` and this is not reflected in the output. That this has slipped by makes me nervous about the whole addition of this PR. Again, `solveset` must carefully analyze the function it is solving and represent a complete solution.
```python
>>> eq=(a/x + exp(x/2)).diff(x, 2)
>>> e1=eq.subs(a,1)
>>> nsolve(e1,-1)
-3.71436772041567
>>> nsolve(e1,-11)
-9.07280730994706
```
This is not a trivial task and should not be rushed. I suggest working away at the ones that you know have a definite solution and conditions sufficient to know the number of solutions.

Reply:
I put this in wolfromalpha I get these roots
![Screenshot from 2019-03-22 18-17-36](https://user-images.githubusercontent.com/39242494/54823803-efb37400-4cce-11e9-8ffc-a3c8c6ec00d3.png)

Can someone please tell me how to use `W-1` . I mean how to get this `W-1` in code

Reply:
Note: this is not the standard for `_tsolve`:
```
There is currently no guarantee that all solutions will be returned or
that a real solution will be favored over a complex one.
```

Reply:
I think the solutions that are missed are complex ones: [2/x**3 + exp(x/2)/4](https://www.wolframalpha.com/input/?i=solve+2%2Fx**3+%2B+exp(x%2F2)%2F4) that is ok, as of now we are only focused in solutions in the real domain.
And with [2*a/x**3 + exp(x/2)/4](https://www.wolframalpha.com/input/?i=solve+2*a%2Fx**3+%2B+exp(x%2F2)%2F4) all the three seems to be same (which is same as the one one returned by solve and solveset)

Reply:
> all the three seems to be same

Are we talking about the same equation? I referenced the two solutions [here](https://github.com/sympy/sympy/pull/14972/files#r267971892).

Reply:
> Are we talking about the same equation?

Yes, I think so, I differentiated the equation: `(a/x + exp(x/2)).diff(x, 2) ->  2*a/x**3 + exp(x/2)/4` and by "all the three seems to be same" I meant all three solutions from [wa](https://www.wolframalpha.com/input/?i=solve+2*a%2Fx**3+%2B+exp(x%2F2)%2F4)

>  I referenced the two solutions

Maybe I didn't understand (or am confused), but `solveset` isn't supposed to give numerical solutions. Isn't it?